### PR TITLE
kickoff: init at v0.7.0

### PR DESCRIPTION
**Pull request by Pyxels**
###### Description of changes

Adds [Kickoff](https://github.com/j0ru/kickoff), which is a minimalistic program launcher for wlroots based compositors with me as a new maintainer.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/kickoff/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+, fontconfig
+, pkg-config
+, wayland
+, libxkbcommon
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "kickoff";
+  version = "v0.7.0";
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:14]:
```suggestion
  version = "0.7.0";
```

File: pkgs/applications/misc/kickoff/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchFromGitHub {
+    owner = "j0ru";
+    repo = pname;
+    rev = version;
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:19]:
```suggestion
    rev = "v${version}";
```

**Review by ocfox — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kickoff/default.nix (hunk 1 above)
Comment (Pyxels) [reply pkgs/applications/misc/kickoff/default.nix]: Done

File: pkgs/applications/misc/kickoff/default.nix (hunk 2 above)
Comment (Pyxels) [reply pkgs/applications/misc/kickoff/default.nix]: Done

File: pkgs/applications/misc/kickoff/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:2]:
```suggestion
```

File: pkgs/applications/misc/kickoff/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "j0ru";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-AolJXFolMEwoK3AtC93naphZetytzRl1yI10SP9Rnzo=";
+  };
+
+  cargoSha256 = "sha256-Twg2C29OwXfCK/rYXnyjbhmCClnsFHz8le9h4AmzXfA=";
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:23]:
```suggestion
  cargoHash = "sha256-Twg2C29OwXfCK/rYXnyjbhmCClnsFHz8le9h4AmzXfA=";
```

File: pkgs/applications/misc/kickoff/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "kickoff";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "j0ru";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-AolJXFolMEwoK3AtC93naphZetytzRl1yI10SP9Rnzo=";
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:20]:
```suggestion
    hash = "sha256-AolJXFolMEwoK3AtC93naphZetytzRl1yI10SP9Rnzo=";
```

File: pkgs/applications/misc/kickoff/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    wrapProgram "$out/bin/kickoff" --prefix LD_LIBRARY_PATH : "${libPath}"
+  '';
+
+  meta = with lib; {
+    description = "Minimalistic program launcher";
+    homepage = "https://github.com/j0ru/kickoff";
+    license = licenses.gpl3Plus;
+    mainProgram = "kickoff";
+    maintainers = with maintainers; [ pyxels ];
+    platforms = [ "x86_64-linux" ];
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:43]:
```suggestion
    platforms = platforms.linux;
```

**Review by ocfox — CHANGES_REQUESTED** [review]
Otherwise LGTM.

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 223553` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kickoff</li>
  </ul>
</details>

File: pkgs/applications/misc/kickoff/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  postInstall = ''
+    wrapProgram "$out/bin/kickoff" --prefix LD_LIBRARY_PATH : "${libPath}"
+  '';
+
+  meta = with lib; {
+    description = "Minimalistic program launcher";
+    homepage = "https://github.com/j0ru/kickoff";
+    license = licenses.gpl3Plus;
+    mainProgram = "kickoff";
```
Comment (ocfox) [inline pkgs/applications/misc/kickoff/default.nix:40]:
```suggestion
```
This is not necessary when only one binary is. Sorry for the many time's suggestions.

Comment (Pyxels) [reply pkgs/applications/misc/kickoff/default.nix]: Ok I removed it. No worries. Thanks for taking the time!

File: pkgs/applications/misc/kickoff/default.nix (hunk 1 above)
Comment (ocfox) [reply pkgs/applications/misc/kickoff/default.nix:13]: This change seems to have been missed.

Comment (Pyxels) [reply pkgs/applications/misc/kickoff/default.nix]: You are totally right, now fixed.

**Review by ocfox — APPROVED** [review]

**Comment by Pyxels** [comment]
Do I need to do anything else for this PR to be merged?

**Review by linyinfeng — APPROVED** [review]
Changes LGTM. Tested basic functionality on my `x86_64-linux` machine with the hyprland compositor.

**Review by NickCao — APPROVED** [review]

**Pull request merged**